core/src/protocols/stream/quic/mod.rs

```diff
@@ -0,0 +1,108 @@
+//! QUIC protocol parser.
+//!
+//! ## Remarks
+//! [QUIC-INVARIANTS] https://datatracker.ietf.org/doc/rfc8999/
+//! [QUIC-RFC9000] https://datatracker.ietf.org/doc/rfc9000/ (Quic V1)
+//! Retina currently only parses Quic Long and Short Headers and does not attempt to parse TLS or HTTP/3 out of
+//! Quic packets. The Quic protocol parser makes several assumptions about the way that quic
+//! packets will behave:
+//! - Assume that the Quic version is one as listed in the QuicVersion Enum in the quic/parser.rs file
+//! - Assume that the dcid of a short header is a maximum of 20 bytes.
+//! - Assume that the packet will not try to grease the fixed bit.
+//!   [QUIC-GREASE](https://www.rfc-editor.org/rfc/rfc9287.html)
+//!
+//! Additionally, there are a couple decisions made in the design of the quic parser:
+//! - The parser will not parse a short header dcid if it is not a part of a pre-identified connection
+//! - The payload bytes count is a lazy counter which does not try to exclude tokens for encryption,
+//!   which is a process that happens in wireshark.
+/*
+TODO: support parsing the tls out of the initial quic packet setup
+TODO support dns over quic
+TODO: support HTTP/3
+*/
+pub(crate) mod parser;
+
+pub use self::header::{QuicLongHeader, QuicShortHeader};
+use serde::Serialize;
+mod header;
+
+/// Parsed Quic Packet contents
+#[derive(Debug, Serialize)]
+pub struct Quic {
```

Comment:
I agree, throughout this implementation, I struggled with naming clarity. I agree changing QuicTransaction is good for clarity, but do you think renaming Quic -> QuicPacket would improve clarity? Because within the other app layer protocols, which have similar implementations to how I've implemented Quic in the Retina library, they all use the name of their protocol within their `mod.rs` files and so changing Quic -> QuicPacket would be slightly inconsistent with the rest of the library.

Reply:
What do you think about QuicTransaction -> QuicHeader for clarity? Because within the usage, you are subscribing to information coming from the quic headers and not getting much information about the payload data (other than an estimate on size).

Reply:
> Because within the other app layer protocols, which have similar implementations to how I've implemented Quic in the Retina library, they all use the name of their protocol 

I agree, but I think Quic is a fundamentally different protocol than TLS, HTTP, etc., because what you're parsing out is more transport protocol-like. None of the other session protocols are delivering data that has a packet-level granularity. 

I like QUIC stream!